nerfstudio/cameras/cameras.py

```diff
@@ -66,67 +70,119 @@ class Cameras:
         distortion_params: OpenCV 6 radial distortion coefficients.
         camera_type: Type of camera model. If a single value is provided, it is broadcasted to
             all cameras. This will be an int corresponding to the CameraType enum.
+        times: Timestamps for each camera
     """
 
+    camera_to_worlds: TensorType["num_cameras":..., 3, 4]
+    fx: TensorType["num_cameras":..., 1]
+    fy: TensorType["num_cameras":..., 1]
+    cx: TensorType["num_cameras":..., 1]
+    cy: TensorType["num_cameras":..., 1]
+    width: TensorType["num_cameras":..., 1]
+    height: TensorType["num_cameras":..., 1]
+    distortion_params: Union[TensorType["num_cameras":..., 6], None]
+    camera_type: TensorType["num_cameras":..., 1]
+    times: Optional[TensorType["num_cameras", 1]]
+
     def __init__(
         self,
-        camera_to_worlds: TensorType["num_cameras", 3, 4],
-        fx: Union[TensorType["num_cameras"], float],
-        fy: Union[TensorType["num_cameras"], float],
-        cx: Union[TensorType["num_cameras"], float],
-        cy: Union[TensorType["num_cameras"], float],
-        width: Optional[Union[TensorType["num_cameras"], int]] = None,
-        height: Optional[Union[TensorType["num_cameras"], int]] = None,
-        distortion_params: Optional[TensorType["num_cameras", 6]] = None,
+        camera_to_worlds: TensorType["batch_c2ws":..., 3, 4],
+        fx: Union[TensorType["batch_fxs":..., 1], TensorType["batch_fxs":...], float],
+        fy: Union[TensorType["batch_fys":..., 1], TensorType["batch_fys":...], float],
+        cx: Union[TensorType["batch_cxs":..., 1], TensorType["batch_cxs":...], float],
+        cy: Union[TensorType["batch_cys":..., 1], TensorType["batch_cys":...], float],
+        width: Optional[Union[TensorType["batch_ws":..., 1], TensorType["batch_ws":...], int]] = None,
+        height: Optional[Union[TensorType["batch_hs":..., 1], TensorType["batch_hs":...], int]] = None,
+        distortion_params: Optional[TensorType["batch_dist_params":..., 6]] = None,
```

Comment:
num_cameras had to go since each fx, fy, height, width, etc all might be of different shapes (and all might be more than one dimension too), in which case we broadcast to the one with the most dims (since it's a TensorDataclass now). They all should be broadcastable is the only constraint